File: java/Basic/Complex/Complex.java
public class Complex {
	private double real;
	private double imaginary;

	public Complex (double real, double imaginary) {
		this.real = real;
		this.imaginary = imaginary; 
	}

	public Complex() {
		this.real = 0;
		this.imaginary = 0;
	}

	public double getReal() {
		return this.real;
	}

	public double getImaginary() {
		return this.imaginary;
	}

	public boolean equal(Complex other) {
		return this.real == other.real && this.imaginary == other.imaginary;
	}

	public boolean notequals(Complex other) {
		return !(this.real == other.real && this.imaginary == other.imaginary);
	}

	public void increment(Complex other) {
		this.real += other.real;
		this.imaginary += other.imaginary;
	}

	public void decrement(Complex other) {
		this.real -= other.real;
		this.imaginary -= other.imaginary;
	}

	public Complex sum(Complex other) {
		Complex result = this;

		result.real += other.real;
		result.imaginary += other.imaginary;

		return result;
	}

	public Complex substr(Complex other) {
		Complex result = this;

		result.real -= other.real;
		result.imaginary -= other.imaginary;
		
		return result;
	}

	public Complex multiple(Complex other) {
		Complex result = this;

		result.real *= other.real;
		result.imaginary *= other.imaginary;

		return result;
	}

	public String toString() {
		return ("(" + this.real + "+" + this.imaginary + "i" + ")");
	}



}

File: dev-club/bcw4/PHP/Complex/Complex.php
<?php

class Complex {
	private $real;
	private $imaginary;

	public function Complex ($real=0, $imaginary=0) {
		$this->real = $real;
		$this->imaginary = $imaginary; 
	}

	public function getReal() {
		return $this->real;
	}

	public function getImaginary() {
		return $this->imaginary;
	}

	public function equal($other) {
		return $this->real == $other->real && $this->imaginary == $other->imaginary;
	}

	public function notequals($other) {
		return !($this->real == $other->real && $this->imaginary == $other->imaginary);
	}

	public function increment($other) {
		$this->real += $other->real;
		$this->imaginary += $other->imaginary;
	}

	public function decrement($other) {
		$this->real -= $other->real;
		$this->imaginary -= $other->imaginary;
	}

	public function sum($other) {
		$result = $this;

		$result->real += $other->real;
		$result->imaginary += $other->imaginary;

		return $result;
	}


	public function substr($other) {
		$result = $this;

		$result->real -= $other->real;
		$result->imaginary -= $other->imaginary;
		
		return $result;
	}

	public function multiple($other) {
		$result = $this;

		$result->real *= $other->real;
		$result->imaginary *= $other->imaginary;

		return $result;
	}

	public function __toString() {
		return sprintf("(%d, %d)", $this->real, $this->imaginary);
	}
}

?>